metadeploy/api/serializers.py

```diff
@@ -223,6 +224,11 @@ def get_not_allowed_instructions(self, obj):
     def get_requires_preflight(self, obj):
         return obj.requires_preflight
 
+    def get_average_duration(self, obj):
+        """Plan.average_duration is an expensive query,
+        so we prefer the already calculated value if available."""
+        return obj.calculated_average_duration or obj.average_duration
```

Comment:
```suggestion
        return obj.calculated_average_duration
```

so that we don't always try to recalculate when calculated_average_duration is None because not enough jobs have run yet